### ISSUE TITLE
Add support for $fill aggregation and $linearFill operator

### ISSUE THREAD

**Issue by francescobianca** [spring-projects/spring-data-mongodb#4327]
Hi, I am using spring data to create aggregations on collection of timeseries type. I would find it useful to use functions such as __$densify__ and __$fill__ for filling gaps in timeseries. Does the latest version of the library support these two native functions?

(Example)
```
db.stock.aggregate( [
   {
      $setWindowFields:
         {
            sortBy: { time: 1 },
            output:
               {
                  price: { $linearFill: "$price" }
               }
         }
   }
] )
```
```
db.weather.aggregate( [
   {
      $densify: {
         field: "timestamp",
         range: {
            step: 1,
            unit: "hour",
            bounds:[ ISODate("2021-05-18T00:00:00.000Z"), ISODate("2021-05-18T08:00:00.000Z") ]
         }
      }
   }
] )
```

**Comment by christophstrobl** [comment]
`DensifyOperation` is already available since 4.0. 
Some of the operators new in Mongo 5.3 are not directly supported yet. Till those are added you can use `Aggregation.stage(...)` to provide the native representation for eg. a `$fill` stage and `AggregationExpression.from(MongoExpression.create("'$linearFill' : '$price'")` for expressions.

**Comment by francescobianca** [comment]
Thanks a lot, I've used the `Aggregation.stage(...)`. It would be helpful if you could pass in an `input.json` file to define the entire pipeline and not the single stage.
